aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -593,5 +594,21 @@ Tensor hann_window(
   return native::hamming_window(
       window_length, periodic, /*alpha=*/0.5, /*beta=*/0.5, options);
 }
+
+template <typename T>
+Tensor tensor(ArrayRef<T> values, const TensorOptions& options) {
+  auto result = at::empty(values.size(), options);
+  for (size_t i = 0; i < values.size(); ++i) {
+    result[i] = values[i];
+  }
+  return result;
+}
+
+#define TENSOR(T, _1, _2)                                           \
+  Tensor tensor(ArrayRef<T> values, const TensorOptions& options) { \
```

Comment:
This is a little weird. It looks like an specialization of the above template, but it's an overload with a concrete type.

Does the templated versions actually buy you anything? I'm worried that it will just make compile errors more confusing.

Reply:
It buys you that you don't have to put the bulk of the code in the macro. It was more useful when the implementation had more code. Now it could probably be put into the macro ...